Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
-            if (m_LastLogMove < 0 && m_InputLogPosition != --(m_InputLog.end()))
-                m_InputLogPosition++;
-
-            // Revive logged input and put cursor at the end of it
-            m_pInputTextBox->SetText(*m_InputLogPosition);
-            m_pInputTextBox->SetCursorPos(m_pInputTextBox->GetText().length());
-            m_InputLogPosition++;
-            m_LastLogMove = 1;
-
-            // Avoid falling off the end
-            if (m_InputLogPosition == m_InputLog.end())
-            {
-                m_InputLogPosition--;
-                m_LastLogMove = 0;
-            }
-        }
-        // Down pressed! Load next logged input string into the input line
-        else if (g_UInputMan.KeyPressed(KEY_DOWN))
-        {
-            // See if we should decrement doubly because the last move was in the opposite direction
-            if (m_LastLogMove > 0 && m_InputLogPosition != m_InputLog.begin())
-                m_InputLogPosition--;
-
-            // If already at the beginning, then only put in empty string
-            if (m_InputLogPosition == m_InputLog.begin())
-            {
-                m_pInputTextBox->SetText("");
-                m_LastLogMove = 0;
-            }
-            else
-            {
-                m_InputLogPosition--;
-                // Revive logged input and put cursor at the end of it
-                m_pInputTextBox->SetText(*m_InputLogPosition);
-                m_pInputTextBox->SetCursorPos(m_pInputTextBox->GetText().length());
-                m_LastLogMove = -1;
-            }
-        }
-    }
-
-// TODO: This sometime.. bleh
-    // Cut off the text in the text label at a reasonable height so it doesn't get really slow to draw
-//    if (m_pConsoleText->GetTextHeight() > g_FrameMan.)
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          Draw
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Draws this ConsoleMan's current graphical representation to a
-//                  BITMAP of choice.
-
-void ConsoleMan::Draw(BITMAP *pTargetBitmap)
-{
-    if (IsVisible() || m_ForceVisible)
-    {
-        AllegroScreen drawScreen(pTargetBitmap);
-        m_pGUIController->Draw(&drawScreen);
-    }
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          ClearLog
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Clears all previous input.
-
-void ConsoleMan::ClearLog()
-{
-    m_InputLog.clear();
-    m_InputLogPosition = m_InputLog.begin();
-    m_pConsoleText->SetText("");
-}
-
-
-} // namespace RTE
+
+		m_InputTextBox->SetText("");
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::ConsoleOpenClose() {
```

Comment:
This is me being nitpicky, but it'd be nice if the method ordering matched up with call order. Right now FeedString and FeedMultipleStrings are called after ConsoleOpenClose but those methods are above it.